src/Data/Profunctor/Join.purs

```diff
@@ -0,0 +1,21 @@
+module Data.Profunctor.Join where
+
+import Prelude
+
+import Data.Functor.Invariant (class Invariant)
+import Data.Newtype (class Newtype)
+import Data.Profunctor (class Profunctor, dimap)
+
+-- | Turns a `Profunctor` into a `Invariant` functor by equating the two type
+-- | arguments.
+newtype Join p a = Join (p a a)
```

Comment:
Yeah, you can derive something Yoneda-esque if you try to find an adjoint to `Join`:

```purescript
data PronedaF i a b x = PronedaF (a -> x) (i x) (x -> b)
type Proneda i a b = Exists (PronedaF i a b)

instance profunctorProneda :: Profunctor (Proneda i) where ...

lowerProneda :: forall i a b. Proneda i a a -> i a
```